opentelemetry-sdk/src/opentelemetry/sdk/trace/sampling.py

```diff
@@ -372,26 +407,52 @@ def __init__(self, rate: float):
 
 
 def _get_from_env_or_default() -> Sampler:
-    trace_sampler = os.getenv(
+    trace_sampler_name = os.getenv(
         OTEL_TRACES_SAMPLER, "parentbased_always_on"
     ).lower()
-    if trace_sampler not in _KNOWN_SAMPLERS:
-        _logger.warning("Couldn't recognize sampler %s.", trace_sampler)
-        trace_sampler = "parentbased_always_on"
 
-    if trace_sampler in ("traceidratio", "parentbased_traceidratio"):
+    if trace_sampler_name in _KNOWN_SAMPLERS:
+        if trace_sampler_name in ("traceidratio", "parentbased_traceidratio"):
+            try:
+                rate = float(os.getenv(OTEL_TRACES_SAMPLER_ARG))
+            except ValueError:
+                _logger.warning(
+                    "Could not convert TRACES_SAMPLER_ARG to float."
+                )
+                rate = 1.0
+            return _KNOWN_SAMPLERS[trace_sampler_name](rate)
+        return _KNOWN_SAMPLERS[trace_sampler_name]
+    else:
         try:
-            rate = float(os.getenv(OTEL_TRACES_SAMPLER_ARG))
-        except ValueError:
-            _logger.warning("Could not convert TRACES_SAMPLER_ARG to float.")
-            rate = 1.0
-        return _KNOWN_SAMPLERS[trace_sampler](rate)
-
-    return _KNOWN_SAMPLERS[trace_sampler]
+            trace_sampler_factory = _import_sampler_factory(trace_sampler_name)
+            sampler_arg = os.getenv(OTEL_TRACES_SAMPLER_ARG, "")
+            trace_sampler = trace_sampler_factory(sampler_arg)
+            if not isinstance(trace_sampler, Sampler):
+                message = (
+                    "Output of traces sampler factory, %s, was not a Sampler object."
+                    % trace_sampler_factory
```

Comment:
Should use f-string here instead of old style

Reply:
Could you explain this further? The warnings I see elsewhere in the code us this format. Furthermore, the linter suggested it when I used something else.

Reply:
You mean the log statements? That would be`logger.warning("foo %s", bar)` vs `logger.warning("foo %s" % bar)` which we have here. It might be good to convert this to the former so structured loggers can see the unformatted value although its duplicating the code.

But for just plain string formatting we use f-strings, see https://github.com/open-telemetry/opentelemetry-python/issues/2130

Reply:
Oh I see. Got it